src/main/java/edu/ucsb/cs48/s20/demo/controllers/StudentsController.java

```diff
@@ -0,0 +1,99 @@
+package edu.ucsb.cs48.s20.demo.controllers;
+
+import edu.ucsb.cs48.s20.demo.entities.Student;
+import edu.ucsb.cs48.s20.demo.repositories.StudentRepository;
+import edu.ucsb.cs48.s20.demo.services.CSVToObjectService;
+import edu.ucsb.cs48.s20.demo.services.MembershipService;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+@Controller
+public class StudentsController {
+
+    private Logger logger = LoggerFactory.getLogger(StudentsController.class);
+
+    @Autowired
+    private MembershipService ms;
+
+    private StudentRepository studentRepository;
```

Comment:
Do we need the `@Autowired` annotation here? The reasoning for why it's needed is still a bit unclear to me, but most of the examples I've seen from the past two quarters include it.

Reply:
No, because it's being set in the constructor. I do think it's good practice to set all the fields in the constructor rather than using the `@Autowired` annotation, though.